### PR TITLE
[Blazor][Wasm] API review feedback

### PR DESCRIPTION
**Pull request by javiercn**

### CODE REVIEW

**Comment by javiercn** [comment]
Applies all the feedback from the API review meeting

File: src/Components/WebAssembly/WebAssembly.Authentication/src/Models/RemoteAuthenticationResult.cs [hunk 1]
```diff
@@ -7,7 +7,7 @@ namespace Microsoft.AspNetCore.Components.WebAssembly.Authentication
     /// Represents the result of an authentication operation.
     /// </summary>
     /// <typeparam name="TState">The type of the preserved state during the authentication operation.</typeparam>
-    public class RemoteAuthenticationResult<TState> where TState : class
+    public class RemoteAuthenticationResult<TState> where TState : RemoteAuthenticationState
```
Comment (pranavkm) [inline src/Components/WebAssembly/WebAssembly.Authentication/src/Models/RemoteAuthenticationResult.cs:10]:
```suggestion
    public class RemoteAuthenticationResult<TRemoteAuthenticationState> where TRemoteAuthenticationState : RemoteAuthenticationState
```

to be consistent with other APIs

**Review by pranavkm — APPROVED** [review]

**Review by captainsafia — APPROVED** [review]

**Comment by mrpmorris** [comment]
@javiercn may I ask what the reasoning was behind changing consts to readonly?

**Comment by javiercn** [comment]
Consts get burned into the target assembly, while static readonly fields don't, so if you update the dependency your assembly picks up the changes even if you don't recompile the app.

**Comment by mrpmorris** [comment]
For the same reason I dislike them then :)

**Pull request merged**